external/storm-solr/src/main/java/org/apache/storm/solr/bolt/SolrUpdateBolt.java

```diff
@@ -92,11 +94,19 @@ private void ack(Tuple tuple) throws SolrServerException, IOException {
         if (commitStgy == null) {
             collector.ack(tuple);
         } else {
-            toCommitTuples.add(tuple);
-            commitStgy.update();
-            if (commitStgy.commit()) {
+            if (TupleUtils.isTick(tuple)) {
+                LOG.debug("TICK! forcing solr client commit");
+                collector.ack(tuple);
+                commitStgy.commit();
                 solrClient.commit(solrMapper.getCollection());
                 ackCommittedTuples();
+            } else {
+                toCommitTuples.add(tuple);
+                commitStgy.update();
+                if (commitStgy.commit()) {
```

Comment:
The strategy in AbstractHdfsBolt is to set a boolean in the case of a tick tuple and then sync if that value is true or if other conditions dictate a sync.  (https://github.com/apache/storm/blob/master/external/storm-hdfs/src/main/java/org/apache/storm/hdfs/bolt/AbstractHdfsBolt.java#L154) 

The benefit of that approach is to eliminate duplicate code (i.e. that calls to ackCommittedTuples() and solrClient.commit()), which I think is a substantial benefit.

Here that would looks something like:

```code
if (forceCommit || commitStgy.commit()) {
  solrClient.commit(solrMapper.getCollection());
  ackCommittedTuples();
}
```
With duplicate code removed I would be +1

A unit test would also be helpful. HdfsBolt example is here: https://github.com/apache/storm/blob/master/external/storm-hdfs/src/test/java/org/apache/storm/hdfs/bolt/TestHdfsBolt.java#L175